samples/grpc/local-drone-control-scala/src/main/scala/local/drones/DroneServiceImpl.scala

```diff
@@ -16,6 +19,8 @@ import scala.concurrent.TimeoutException
 
 class DroneServiceImpl(
     deliveriesQueue: ActorRef[DeliveriesQueue.Command],
+    chargingStationEntityRefFactory: String => EntityRef[
+      ChargingStation.Command],
```

Comment:
Why do we need the indirection of this factory function? It could use `ClusterSharding(system).entityRefFor`?

Reply:
It's from the `Replication#entityRefFactory`/`EdgeReplication#entityRefFactory` since the user doesn't set up sharding for the entity themselves and so it might not be obvious that it is a side effect of starting it with `Replication.grpcReplication` or `Replication.grpcEdgeReplication` that there is sharding with the entity type running after that.

Reply:
But maybe it would be more clear/clean to pass the entire `EdgeReplication`?

Reply:
ah, I see that comes from the `EdgeReplication.entityRefFactory`, and we have the same in the other grpcReplication